elastic/kibana
# [Infra UI] Add Features section to Infra settings

Closes https://github.com/elastic/kibana/issues/167062

## Summary

* Adds "Features" section to Infra settings screen
* Hooks changes to `uiSettings` into existing form in order to use the current "Discard" and "Apply" buttons logic
* Replaces usage of `uiSettings` with `settings.client` as `uiSettings` seems to be [deprecated](https://github.com/elastic/kibana/pull/154710. Observables exposed by `uiSettings` do not seem to work anymore.
* Deletes the Hosts landing screen
* Revives the logic for showing/hiding Hosts nav item depending on the settings toggle

https://github.com/elastic/kibana/assets/793851/767e69cc-98b1-4fe9-93d9-f39c08ac5c83


## How to test

* Checkout locally and run in stateful mode
* Go to Infra Setttings screen and make sure Features section is present
* Make sure that the "Infrastructure Hosts view" toggle shows/hides the Hosts nav item
* Make sure that "Infrastructure Hosts view" in "Anvanced Settings" is synced with the the one in Infra settings
